Replace Go tests with BATS and remove Go dependency

jnovy opened the pull request:
Reimplemented all tests from Go to BATS framework, providing comprehensive test coverage while eliminating Go build dependencies. Added 50 BATS tests covering basic functionality, container logging, k8s log rotation, and full runtime integration with real container execution.

Fixes: #577

haircommander commented on test/02-ctr-logs.bats:
```diff
@@ -0,0 +1,79 @@
+}
+
+@test "ctr logs: log driver as journald with short cid should fail" {
+    local short_ctr_id="abcdefghijkl"
```
actually a number of these do

jnovy commented:
Thanks @haircommander for the review - the two tests were indeed duplicated.

Duplicate Tests Fixed:
  - Removed duplicate "log driver as journald with short cid should fail" test from test/02-ctr-logs.bats (kept the original in test/01-basic.bats)
  - Removed duplicate "log driver as k8s-file with path should pass" test from test/02-ctr-logs.bats (kept the original in test/01-basic.bats)

Seems no remaining files import or reference the deleted modules (runner/config, runner/conmon, cmd/conmon-config).

haircommander commented:
LGTM, thanks!
